pycbc/distributions/sky_location.py

```diff
@@ -161,18 +159,244 @@ def rvs(self, size):
+
+            margin = 2 * numpy.pi / (4 * nside - 1)
+
+            delta_max = min(delta_max + margin, numpy.pi / 2)
+            delta_min = max(delta_min - margin, -numpy.pi / 2)
+            alpha_max = min(alpha_max + margin, 2 * numpy.pi)
+            alpha_min = max(alpha_min - margin, 0)
+
+            return delta_min, delta_max, alpha_min, alpha_max
+
+        file_name = params['healpix_file']
+
+        coverage = params['coverage']
+
+        if coverage > 1 or coverage < 0:
+            raise ValueError(
+                f'Coverage must be between 0 and 1, {coverage} is not correct'
+            )
+
+        rasterization_nside = params['rasterization_nside']
+
+        if bin(rasterization_nside).count('1') != 1:
+            raise ValueError(
+                f'Rasterization_nside must be a power of 2,'
+                f'{rasterization_nside} is not correct'
+            )
+        self.healpix_map = mhealpy.HealpixMap.read_map(file_name)
+        self.boundaries = boundaries(
+            self.healpix_map, rasterization_nside, coverage
+        )
+        logging.info('HealpixSky boundary is %s', self.boundaries)
+
+    @property
+    def params(self):
+        return self._params
+
+    @classmethod
+    def from_config(cls, cp, section, variable_args):
+        tag = variable_args
+        variable_args = variable_args.split(VARARGS_DELIM)
+        if set(variable_args) != set(cls._params):
+            raise ValueError(
+                "Not all parameters used by this distribution "
+                "included in tag portion of section name"
+            )
+        healpix_file = str(cp.get_opt_tag(section, 'healpix_file', tag))
+        coverage = 0.9999
+        if cp.has_option_tag(section, 'coverage', tag):
+            coverage = float(cp.get_opt_tag(section, 'coverage', tag))
+
+        rasterization_nside = 64
+        if cp.has_option_tag(section, 'rasterization_nside', tag):
+            rasterization_nside = int(
+                cp.get_opt_tag(section, 'rasterization_nside', tag)
+            )
+        return cls(
+            healpix_file=healpix_file,
+            coverage=coverage,
+            rasterization_nside=rasterization_nside,
+        )
+
+    def rvs(self, size):
+        def simple_rejection_sampling(healpix_map, size, boundaries):
+            """Start from a uniform distribution of points, and accepts those
+             whose values on the map are greater than a random value
+             following a uniform law
+
+             Parameters
+             ----------
+             healpix_map  : HealpixMap instance
+             size : int
+                 number of points tested by the method
+             boundaries : list -> tuple of 4 floats
+                 delta_min,delta_max,alpha_min,alpha_max = boundaries
+                 delta is the declination in radians [-pi/2, pi/2]
+                 alpha is the right ascention in radians [0,2pi]
+
+             Returns
+             -------
```

Comment:
Something is funny with the indentation here. Perhaps this is what codeclimate raised, but I haven't checked.